dashboard/app/models/pd/enrollment.rb

```diff
@@ -159,12 +157,15 @@ def self.filter_for_survey_completion(enrollments, select_completed = true)
     raise 'Expected enrollments to be an Enumerable list of Pd::Enrollment objects' unless
         enrollments.is_a?(Enumerable) && enrollments.all?(Pd::Enrollment)
 
-    # Local summer, CSP Workshop for Returning Teachers, or CSF Intro after 5/8/2020 will use Foorm for survey completion.
-    # CSF Deep Dive after 9/1 also uses Foorm. CSF District workshops will always use Foorm
+    # Filter out Local summer, CSP Workshop for Returning Teachers, and CSF Intro workshops before 5/8/2020 (they
+    # do not use Foorm for survey completion); CSF Deep Dive workshops before 9/1/2020 (they do not use Foorm for
+    # survey completion); and Admin + Admin/Counselor workshops (they should not receive exit surveys at all).
     foorm_enrollments = enrollments.select do |enrollment|
-      (enrollment.workshop.workshop_ending_date >= Date.new(2020, 5, 8) &&
-        (enrollment.workshop.csf_intro? || enrollment.workshop.local_summer? || enrollment.workshop.csp_wfrt?)) ||
-        (enrollment.workshop.workshop_ending_date >= Date.new(2020, 9, 1) && enrollment.workshop.csf_201?) || enrollment.workshop.csf_district?
+      enrollment.workshop.course != Pd::Workshop::COURSE_ADMIN &&
+      enrollment.workshop.course != Pd::Workshop::COURSE_ADMIN_COUNSELOR &&
+      !(enrollment.workshop.workshop_ending_date < Date.new(2020, 5, 8) &&
+        (enrollment.workshop.csf_intro? || enrollment.workshop.local_summer? || enrollment.workshop.csp_wfrt?)) &&
```

Comment:
I'm confused here –– it looks like the `(enrollment.workshop.csf_intro? || enrollment.workshop.local_summer? || enrollment.workshop.csp_wfrt?)` is part of the not (`!`) condition, but I don't see the `!` being applied to these in the statement above –– what am I missing?

Reply:
Oh those 3 are part of the (!) condition that makes sure they weren't for before that date (8/5/2020) since before then we didn't use Foorm for it. The overall conditional is checking to make sure the given workshop is/was not A) an admin workshop nor a admin/counselor workshop, B) ended before 8/5/2020 if it is a CSF Intro, local summer, or CSP WFRT workshop, nor C) ended before 9/1/2020 if it is a CSF 201 workshop. So the admin + admin/counselor workshops are not dependent on dates, they should just always be filtered out.

Reply:
I could pull out the conditionals and store them in variables like putting all the date-related ones in one variable and having the 2 admin ones in another and then using those to write the "select" conditionals? That may make it more readable!

Reply:
I see! That makes sense. I do think creating small methods / variables here would help so there are fewer conditions to track, but thanks for the explanation!